src/engine.ml

```diff
@@ -1274,30 +1276,53 @@ let check_control_integrity err key p hmac_algorithm hmac_key =
   let+ () = guard (Cstruct.equal computed_mac packet_mac) (err computed_mac) in
   Log.info (fun m -> m "mac good")
 
-let split_control mtu outs =
-  List.concat_map
-    (function
-      | `Control, data ->
-          let rec datas acc data =
-            if Cstruct.is_empty data then List.rev acc
-            else
-              let l = min mtu (Cstruct.length data) in
-              let data, data' = Cstruct.split data l in
-              datas (data :: acc) data'
-          in
-          List.map (fun data -> (`Control, data)) (datas [] data)
-      | ((`Ack | `Reset_server | `Reset), _) as p -> [ p ])
-    outs
+let split_control ~acks mtu outs =
+  (* only the first control/ack packet is carrying acks *)
+  List.rev
```

Comment:
I think you forgot to remove this one in the latest commit :-)

Reply:
Hmm? we're folding left, and accumulate "new_thing :: acc", so in the end we want to reverse to retain the order.

I removed in [62c3a61](https://github.com/robur-coop/miragevpn/pull/162/commits/62c3a6198e1dfd72b046210b0fffccea20bdd521) the inner List.rev when we actually split some control data.

Reply:
Ah my bad